src/test/java/com/amihaiemil/camel/NoCommentsYamlLineTest.java

```diff
@@ -0,0 +1,121 @@
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+package com.amihaiemil.camel;
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Unit test for {@link NoCommentsYamlLine}.
+ * @author Sherif Waly (sherifwaly95@gmail.com)
+ * @version $Id$
+ * @since 1.0.1
+ */
+public final class NoCommentsYamlLineTest {
+    /**
+     * NoCommentsYamlLine knows its number.
+     */
+    @Test
+    public void knowsNumber() {
+        YamlLine line = new NoCommentsYamlLine(new RtYamlLine("line", 11));
+        MatcherAssert.assertThat(line.number(), Matchers.is(11));
+    }
+    
+    /**
+     * NoCommentsYamlLine returns its, indentation if it's right.
+     */
+    @Test
+    public void isWellIndented() {
+        YamlLine byZero = new NoCommentsYamlLine(
+            new RtYamlLine("this: line", 5)
+        );
+        MatcherAssert.assertThat(
+            byZero.indentation(), Matchers.is(0)
+        );
+        YamlLine byFour = new NoCommentsYamlLine(
+                new RtYamlLine("    this: line", 10)
+        );
+        MatcherAssert.assertThat(
+            byFour.indentation(), Matchers.is(4)
+        );
+    }
+    
+    /**
+     * NoCommentsYamlLine compares right with other lines.
+     */
+    @Test
+    public void comparesRight() {
+        YamlLine noComments = new NoCommentsYamlLine(
+            new RtYamlLine("abc", 1)
+        );
+        YamlLine normalLine = new RtYamlLine("  no", 2);
+        MatcherAssert.assertThat(
+            noComments.compareTo(normalLine) < 0, Matchers.is(true)
+        );
+    }
+    
+    /**
+     * NoCommentsYamlLine trims a comment line and return empty line.
+     */
+    @Test
+    public void trimsCommentsLine() {
+        YamlLine noComments = new NoCommentsYamlLine(
+            new RtYamlLine("   #this is a comment line", 1)
+        );
+        MatcherAssert.assertThat(
+            noComments.trimmed(), Matchers.is("")
+        );
+    }
+    
+    /**
+     * NoCommentsYamlLine removes only comments and leaves original string.
+     */
+    @Test
+    public void trimsCommentsAtEndOfLine() {
+        YamlLine noComments = new NoCommentsYamlLine(
+            new RtYamlLine("  this isn't comment   #here is the comment", 1)
+        );
+        MatcherAssert.assertThat(
+            noComments.trimmed(), Matchers.is("this isn't comment")
+        );
+    }
+    
+    /**
+     * NoCommentsYamlLine doesn't remove # escaped in a string.
+     */
+    @Test
+    public void doesnotTrimsEscapedHash() {
```

Comment:
The name here should be ``doesNotTrimEscapedHash()`` (N and Trim instead of Trims)